iridauploader/core/upload_helpers.py

```diff
@@ -225,7 +225,10 @@ def upload_sequencing_run(sequencing_run, directory_status, upload_mode, upload_
         logging.error("Lost connection to Irida")
         logging.error("Errors: " + pformat(e.args))
         full_error = "Lost connection to Irida. Errors: " + pformat(e.args)
-        _set_and_write_directory_status(directory_status, DirectoryStatus.ERROR, full_error)
+        if directory_status.run_id is not None:  # A run ID was generated, so the run is partial and can be continued
+            _set_and_write_directory_status(directory_status, DirectoryStatus.PARTIAL, full_error)
```

Comment:
I wonder if this logic should instead be moved to the `_set_and_write_directory_status` function. The reason I think this is because I have been reviewing the rest of the code to understand context and I see a lot of places where it catch's exception and if those other places are hit when the run was initially PARTIAL it will force re-upload of everything. I.E. `irida_prep_and_validation`.
From what I can tell run status get's set to `PARTIAL` when it begins and it would stay in that state if the uploader crashes before completion and this new case where it has a connection error when starting upload. What about when the second case happens and when it hit's `parse_and_validate` and encounters an exception on the secondary execution. Samples are still partially uploaded but they will now be forced to re-upload. (i.e. user loses internet connection on the uploader)

Reply:
That's a good idea I have changed it now. I also noticed an edge case where a run could be set to PARTIAL before the run_id is set, so I changed it so that only _set_and_write_directory_status will set a run to PARTIAL